src/hotspot/os_cpu/aix_ppc/thread_aix_ppc.cpp

```diff
@@ -54,9 +55,19 @@ bool JavaThread::pd_get_top_frame_for_profiling(frame* fr_addr, void* ucontext,
   if (has_last_Java_frame() && frame_anchor()->walkable()) {
     intptr_t* sp = last_Java_sp();
     address pc = _anchor.last_Java_pc();
-    // pc can be seen as null because not all writers use store pc + release store sp.
-    // Simply discard the sample in this very rare case.
-    if (pc == nullptr) return false;
+    if (pc == nullptr) {
+      // This is not uncommon. Many c1/c2 runtime stubs do not set the pc in the anchor.
+      intptr_t* top_sp = os::Aix::ucontext_get_sp((const ucontext_t*)ucontext);
+      if ((uint64_t)sp <= ((frame::abi_minframe*)top_sp)->callers_sp) {
+        // The interrupt occurred either in the last java frame or in its direct callee.
+        // We cannot be sure that the link register LR was already saved to the
+        // java frame. Therefore we discard this sample.
+        return false;
+      }
+      // The last java pc will be found in the abi part of the last java frame.
+      *fr_addr = frame(sp);
```

Comment:
I'm not a big fan of this diff, but the alternatives (changing frame constructor or determining pc here) are probably not better. So, ok.